### PR TITLE
ElementQuery media polyfill.

### PR DESCRIPTION
**Pull request by vogdb**
Experimental issue to try http://www.smashingmagazine.com/2013/06/25/media-queries-are-not-the-answer-element-query-polyfill/.

### CODE REVIEW

**Comment by vogdb** [comment]
@christopheschwyzer please look and test how it looks. Is it acceptable for us?

**Comment by christopheschwyzer** [comment]
Looks good. I like it. 
- I would use the same media-query breakpoint (@breakpointSmall). Also I noticed that this variable should be defined in CM. 
- We should use the same behavior in _form.less_ for the form layout

**Comment by vogdb** [comment]
> I would use the same media-query breakpoint (@breakpointSmall). Also I noticed that this variable should be defined in CM.

So, should I define it if it is not defined?

**Comment by christopheschwyzer** [comment]
> Should I define all the breakpoint variables? 

Yes, we should define the variables in CM, if they are used in CM..

**Comment by christopheschwyzer** [comment]
IMO you can move all the breakPoint variables...

**Comment by vogdb** [comment]
Hm. According to http://leafo.net/lessphp/changelog.html
> - Selector interpolation works inside of brackets in selector

works since 0.4. We have 0.3.9. So, my question is. Can we update `lessphp` to 0.4.0? @njam please answer.

**Comment by njam** [comment]
I think we should update!
There's even 0.5.0 on github: https://github.com/leafo/lessphp/releases

You wanna try?

**Comment by vogdb** [comment]
Yes, please!

**Comment by vogdb** [comment]
Should I just update the dependency on composer.json of CM to?
```json
"leafo/lessphp": "0.5.*",
```

**Comment by njam** [comment]
Yes, and also test that the CSS in SK still compiles properly!
Also let's switch to `~0.5.0'` - https://getcomposer.org/doc/01-basic-usage.md#next-significant-release-tilde-and-caret-operators-

**Comment by vogdb** [comment]
@njam @christopheschwyzer please review.

**Comment by vogdb** [comment]
@christopheschwyzer Should I remove an issue in SK for removal of `breakpoint` variables there because now we have them in CM.

File: layout/default/css/tableField.less [hunk 1]
```diff
@@ -1,29 +1,28 @@
...
+  > .label {
+    box-sizing: border-box;
+    font-weight: bold;
+    color: @fontColorStrong;
 
-  @media (min-width: @breakpointSmall) {
-    width: 100px;
-    float: left;
-    text-align: right;
+    :after {
```
Comment (christopheschwyzer) [inline layout/default/css/tableField.less]: You need "&" for this to work

File: layout/default/css/tableField.less [hunk 2]
```diff
@@ -1,29 +1,28 @@
 .tableField {
   padding: 5px 0 5px 5px;
-}
 
-.tableField:first-child {
-  border-width: 0;
-}
+  :first-child {
```
Comment (christopheschwyzer) [inline layout/default/css/tableField.less]: You need "&" for this to work

**Comment by christopheschwyzer** [comment]
> Should I create an issue in SK for removal of breakpoint variables

Yes - or you can do it in https://github.com/cargomedia/sk/issues/1218

**Comment by vogdb** [comment]
@christopheschwyzer please review.

**Comment by christopheschwyzer** [comment]
Can you also remove 
```
  &:first-child {
    border-width: 0;
  }
```
Doesn't do anything. And can you use
```
.tableField {
 padding: 6px;
}
```
so it's streamlined with form fields...

**Comment by vogdb** [comment]
Yes, Master!

**Comment by vogdb** [comment]
@christopheschwyzer please look

**Comment by christopheschwyzer** [comment]
Lgtm!
Restarted tests.

**Comment by vogdb** [comment]
@njam please review my last commit @3a4bd50

**Comment by njam** [comment]
- function.lessVariable.php: Why did you remove the check of the return value of `preg_match`? I would keep it for clarity?

**Comment by vogdb** [comment]
Because now the lessphp lib throws an exception in case if a variable is not defined. Previously it didn't throw and the check did this throw instead of lessphp. Should I return the check?

**Comment by njam** [comment]
I think it should still be there, as it's a different error condition: Basically if the regexp doesn't match. This could happen for different reasons - e.g. if the returned CSS is not what we expect.

**Comment by vogdb** [comment]
@njam please look.

**Comment by njam** [comment]
Looks good.

Did you do some benchmarking to understand the performance impact of adding this library?

**Comment by vogdb** [comment]
Nope. Will do. Do we have a company standard for benchmarking?

**Comment by njam** [comment]
This case is quite tricky. I would maybe measure the time required to rotate/resize the window on a slow phone?

**Comment by vogdb** [comment]
lessphp 5.0

|Type of device | Load time | Rotate time |
|--------------------|---------------|-----------------|
| Emulator 4.1.2 | 0:50, 0:56, 2:25, 2:02. After browser restart 0:30, 0:34 | ~0:05-0:06 |
| LG p500 4.0.4  | 0:30, 0:33, 0:35 | ~0:05-0:06 |


lessphp 3.9

|Type of device | Load time | Rotate time |
|--------------------|---------------|-----------------|
| Emulator 4.1.2 | 1:34, 1:40 After browser restart 0:33, 0:30 | ~0:05-0:06 |
| LG p500 4.0.4  | 0:35, 0:30 | ~0:05-0:06 |

Conclusion: Android devices are so unpredictable, slow and unstable that it is hard to measure anything serious there. Althought with lessphp 5.0 it seems to me that I saw a more continuous presence of this state:

**Comment by njam** [comment]
Just to be sure: I meant what the impact of the *ElementQueries* plugin is.
I guess the 3.9 version is without and the 5.0 version is with ElementQueries?

If so then it looks like it doesn't make much difference, and I'd say it's good to go?

**Comment by vogdb** [comment]
> I guess the 3.9 version is without and the 5.0 version is with ElementQueries?

Yep.

**Comment by vogdb** [comment]
Ok. @christopheschwyzer @njam if you don't have any more comments then I'm merging this.

**Comment by christopheschwyzer** [comment]
GoGoGo

**Comment by njam** [comment]
⁠

**Pull request merged**